src/Services/ProductRegistrationService.php

```diff
@@ -26,41 +28,71 @@ public function __construct()
     {
     }
 
-    public function getProductStatus()
+    public function getProductStatus(): array
     {
         $row = sqlQuery("SELECT * FROM `product_registration`");
-
         if (empty($row)) {
             $row = [];
         }
-
-        $email = $row['email'] ?? '';
+        $email = $row['email'] ?? null;
+        $lastAskVersion = $row['last_ask_version'] ?? '';
         $optOut = $row['opt_out'] ?? null;
         $telemetry_disabled = $row['telemetry_disabled'] ?? null;
 
-        if (empty($row) || $optOut == null) {
-            $row['statusAsString'] = 'UNREGISTERED';
-        } elseif (!empty($email)) {
-            $row['statusAsString'] = 'REGISTERED';
-        } elseif ($optOut == 1) {
-            $row['statusAsString'] = 'OPT_OUT';
-        }
-
-        $row['allowTelemetry'] = $telemetry_disabled === null ? 1 : null;
+        $row['allowEmail'] = 0;
+        $row['allowTelemetry'] = 0;
         $row['allowRegisterDialog'] = 0;
-        if ($telemetry_disabled == null || $optOut == null) {
-            $row['allowRegisterDialog'] = 1;
+        $currentVersion = (new VersionService())->asString();
+        if ($currentVersion != $lastAskVersion) {
+            // Change in version (or empty entry), so ignore opt outs and show the dialog if empty email or telemetry not enabled
+            if (empty($email) || $telemetry_disabled == 1 || $telemetry_disabled == null) {
+                $row['allowRegisterDialog'] = 1;
+                if (empty($email)) {
+                    $row['allowEmail'] = 1;
+                }
+                if ($telemetry_disabled == 1 || $telemetry_disabled == null) {
+                    $row['allowTelemetry'] = 1;
+                }
+            }
+        } else {
+            // No change in version, so do not show the dialog if has opted out
+            if ($telemetry_disabled == null || $optOut == null) {
+                $row['allowRegisterDialog'] = 1;
+                if ($optOut == null) {
+                    $row['allowEmail'] = 1;
+                }
+                if ($telemetry_disabled == null) {
+                    $row['allowTelemetry'] = 1;
+                }
+            }
         }
```

Comment:
This didn't include if telemetry_disabled is 0 meaning enabled.
Was not permitting recording menu clicks in open tab routines.
```
if (top.allowTelemetry) {
    reportMenuClickData(data);
 }
```
Still unsure if  logic was changed elsewhere but will look more closely in my PR.

I'll fix in my PR that's for adding new events from portal.

Reply:
Thanks for bringing this in. Yep, i was going to change those settings to allowTelemetryModal etc to clarify that they are for showing the modal, which then would of made me realize that the setting for actually running telemetry would be different. I'd prob just make another variable in the  return array called runTelemetry or something like that which can be used to set the top.allowTelemetry (would prob also rename that to runTelemetry or something like that to ensure no confusion) setting in main.php